### DISCUSSION TITLE
4주차 금요일 미팅

### DISCUSSION THREAD

**Discussion by KingNono1030** [DevFDev/frontend#116] [회의록]
## 근황

- 용빈: 치킨 먹을 예정
- 아영: 피자, 옥수수 감자칩 피자
- 윤호: 블랙 타이거 쉬림프 피자

### 했던거
윤호
- 설렉트 컴포넌트 (기본)

용빈
- 공통 컴포넌트 (스위치, 텍스트애리아) -> 피드백 반영, 완성
  - 스위치를 하며, 접근성 관련 공부 !
- 만드니까 공부가 잘 됨
- form 준비 완료

아영
- 프록시 어떻게 할지 감이 오지 않았는데
  - 드디어 이해 완료 -> 로그인, 로그아웃 프록시 사용해서 처리
  - 코드리뷰, api 분리, post 호출
  - axios 안 써도 되서
- 지금은 페이지에서 직접 호출, 함수의 분리 없이
  - 이 함수의 분리를 굳이 해야할까?

@KingNono1030 
- 만약에 분리하면 백에 보내는 api / next 에 보내는 api 구분
  - services / user, team, portfolio ---> 우리 프로젝트 api
  - ---> Next api routes ------> 우리 프로젝트 api

1.  services > auth 
-> 폴더 (routes ?)
- 얘네는 경로가 api routes 로 보내는 구나 (알음알음)
- 명시적으로 알 수는 없지만, 복잡도를 낮춘다

2.  services, routes > auth
- 경로가 다른 api 요청 
- 디렉토리 자체에서 다른 api 요청이라는 힌트

- services 에 두긴 하고, 분리를 안 하면 (loginform > 이 위에서)

@yellowjang 
- 이미 src 아래 폴더가 많은데, 더 이상 늘어나는거 원치 않아
- 서비스 > auth 에 로그인 관리 / 회원가입, 로그인, 로그아웃 정도 -> 단순한 구조로도 정리 가능!

@yongb2n 
- 같은 생각
- 폴더 구조 -> 사람끼리의 약속, 정답 없음
  - 애매하긴 함 (멘탈 모델이라) ---> 그래서 국룰 fsd 가 인기
  - 유동적으로 가져가야
  - fsd 이것도 엄숙하게 숙지하기 어려워

폴더 구조 관련
services > team > index.ts (team 관련 crud, 기본 get post delete 등...)
services > team > index.ts (crud, 기본 get post delete 등...)

queries > team > index.ts (team 관련 crud, 기본 get post delete 등...)
queries > team > index.ts (crud, 기본 get post delete 등...)

1. services 안 쓰고 queries > 안에서 api method + 쿼리문
2. services 에서는 api method / queries 에서 얘네들 import 해서 쿼리문

일단은 분리하고 -> 나중에 합치기
-> 어차피 api 메서드, 쿼리에서 밖에 안 쓰임. 모든 요청이 리액트 쿼리 거쳐야 할 필요 -> 

### 할 거
윤호
- 유저 타입 + 스웨거 업데이트 -> open api typescript -> 최신화
- 설렉트 -> 멀티 설렉트, 멤버 설렉트
- 체크박스 -> css
- assets (아이콘 추가) select, dropdown

용빈
- 폼 작업 -> useFormContext -> context + compound (드롭다운, 모달, form)

아영
- api 메서드 분리
- 미들웨어로 리프레시 토큰 관리
- 회원가입 (-> 인가된 X, api route X)

## 야망
아영
- 텍스트 에디터 정복
용빈
- 훅 폼 정복
윤호

## 취업준비 스터디 -> 리포지토리 -> 공유 노션 (아영/윤호/용빈 -> )
- 방어 기준 벌금 (아영: 2천원 / 윤호: 3천원 / 용빈: 5천원)
- 휴가권 (1주일에 한 번)
- 기본 매일 한다고 생각 (일주일에 5번)
- 담주 월요일 아침 -> 스터디 이야기 진행,
  - 화요일부터 시작
  - 같은 문제 -> 따로 풀고 공유
- 벌금 걸고 스터디
- 이력서 피드백 -> 포트폴리오 -> 리드미
- cs 공부, 코테 준비, 이력서 작성

**Comment by yongb2n** [comment]
벌금 0원 기원...!

![image](https://github.com/user-attachments/assets/9101bd7a-a19f-4cc3-9959-a83de3e838c8)
